docs/OpenTypeFeatureFileSpecification.md

```diff
@@ -0,0 +1,3769 @@
+tell the OT layout engine simply to match the specified sequences, and not to
+perform any substitutions on them. As a result of the match, remaining rules
+(i.e. subtables) in the lookup will be skipped when the rule matches. (See the
+OT layout algorithm in §[7.a](#7.a).)
+
+###### Example 1. Ignoring specific sequences: The `ignore substitute` rules
+below will block any subsequent rules that specifies a substitution for “d” when
+the context around “d” matches any of the sequences “f a d”, “f e d”, or “a d
+d”.
+
+Note that the marked glyphs in the exception sequences indicate where a
+substitution would have occurred; this is necessary for the OpenType layout
+engine to correctly handle skipping this sequence.
+
+```fea
+ignore substitute f [a e] d';
+ignore substitute a d' d;
+substitute [a e n] d' by d.alt;
+```
+
+###### Example 2.
+Matching a beginning-of-word boundary:
+
+```fea
+ignore substitute @LETTER f' i';
+substitute f' i' by f_i.begin;
+```
+
+The example above shows how a ligature may be substituted at a word boundary.
+@LETTER must be defined to include all glyphs considered to be part of a word.
+The substitute statement will get applied only if the sequence doesn't match
+"@LETTER f i"; i.e. only at the beginning of a word.
+
+###### Example 3.
+Matching a whole word boundary:
+
+```fea
+ignore substitute @LETTER a' n' d', a' n' d' @LETTER;
+substitute a' n' d' by a_n_d;
+```
+
+In this example, the a_n_d ligature will apply only if the sequence “a n d” is
+neither preceded nor succeeded by a @LETTER. Also, note the use of the comma in
+the `ignore` statement: this is equivalent to writing:
+
+```fea
+ignore substitute @LETTER a' n' d';
+ignore substitute a' n' d' @LETTER;
+substitute a' n' d' by a_n_d;
+```
+
+###### Example 4.
+This shows a specification for the contextual swashes feature:
+
+```fea
+feature cswh {
+
+    # --- Glyph classes used in this feature:
+    @BEGINNINGS = [A-N P-Z Th m];
+    @BEGINNINGS_SWASH = [A.swash-N.swash P.swash-Z.swash T_h.swash m.begin];
+    @ENDINGS = [a e z];
+    @ENDINGS_SWASH = [a.end e.end z.end];
+
+    # --- Beginning-of-word swashes:
+    ignore substitute @LETTER @BEGINNINGS';
+    substitute @BEGINNINGS' by @BEGINNINGS_SWASH;
+
+    # --- End-of-word swashes:
+    ignore substitute @ENDINGS' @LETTER;
+    substitute @ENDINGS' by @ENDINGS_SWASH;
+
+} cswh;
+```
+
+If a feature only targets glyphs at the beginning or ending of a word, such as
+the `init` and `fina` features, then the application could be made responsible
+for detecting the word boundary; the feature itself would be simply defined as
+the appropriate substitutions without regard for word boundary. Such application
+responsibilities must be described in the feature tag registry.
+
+<a name="5.g"></a>
+### 5.g. [GSUB LookupType 7] Extension substitution
+
+The `useExtension` keyword specifies creating lookups of this lookup type. See
+§[4.e](#4.e) and
```

Comment:
add link https://docs.microsoft.com/en-us/typography/opentype/spec/os2#ur

Reply:
Cannot act on this because wrong preview lines are shown (as mentioned above).